pkg/collectcfg/fetch.go

```diff
@@ -200,6 +200,13 @@ func PullPodFiles(podId string, containerName string, remotePath string, localPa
 }
 
 func SyncConfigDir(localPath string, remotePath string, sshCmd string, undercloud string) error {
+	// make sure localPath exists
+	mkCmd := "mkdir -p " + localPath
```

Comment:
I don't like this, hit's not healthy to Exec or shell out when Go has purpose made idioms.  If we can avoid shelling out we get two benefits among others:

1) We are OS agnostic, the lower level functions will work in different environments from Windows to macOS to BSD to any form of Linux
2) We are not at the mercy of potential security vulnerabilities by blinding trusting external tools on our operator's machine

Example 1:

`os.Mkdirall() // https://pkg.go.dev/os#MkdirAll`

Example 2:

We could make a function:
```
func createDirectory(path string) error {
   // Check if the directory exists
   _, err := os.Stat(path)
   if err == nil {
      return nil
   }
   // If the directory does not exist, create its parent
   if os.IsNotExist(err) {
      err = createDirectory(filepath.Dir(path))
      if err != nil {
         return err
      }
      // Create the directory
      err = os.Mkdir(path, 0777)
      if err != nil {
         return err
      }
   }
   return nil
}```

Reply:
Thank you for the review, and yes os.Mkdirall() is a best approach.
Fixed in the next commit